contracts/GenArt721CoreV3_Engine.sol

```diff
@@ -0,0 +1,2038 @@
+        if (artistRevenue_ > 0) {
+            artistAddress_ = projectFinance.artistAddress;
+        }
+        if (additionalPayeePrimaryRevenue_ > 0) {
+            additionalPayeePrimaryAddress_ = projectFinance
+                .additionalPayeePrimarySales;
+        }
+    }
+
+    /**
+     * @notice Backwards-compatible (pre-V3) getter returning contract admin
+     * @return address Address of contract admin (same as owner)
+     */
+    function admin() external view returns (address) {
+        return owner();
+    }
+
+    /**
+     * @notice Gets the project ID for a given `_tokenId`.
+     * @param _tokenId Token ID to be queried.
+     * @return _projectId Project ID for given `_tokenId`.
+     */
+    function tokenIdToProjectId(uint256 _tokenId)
+        public
+        pure
+        returns (uint256 _projectId)
+    {
+        return _tokenId / ONE_MILLION;
+    }
+
+    /**
+     * @notice Convenience function that returns whether `_sender` is allowed
+     * to call function with selector `_selector` on contract `_contract`, as
+     * determined by this contract's current Admin ACL contract. Expected use
+     * cases include minter contracts checking if caller is allowed to call
+     * admin-gated functions on minter contracts.
+     * @param _sender Address of the sender calling function with selector
+     * `_selector` on contract `_contract`.
+     * @param _contract Address of the contract being called by `_sender`.
+     * @param _selector Function selector of the function being called by
+     * `_sender`.
+     * @return bool Whether `_sender` is allowed to call function with selector
+     * `_selector` on contract `_contract`.
+     * @dev assumes the Admin ACL contract is the owner of this contract, which
+     * is expected to always be true.
+     * @dev adminACLContract is expected to either be null address (if owner
+     * has renounced ownership), or conform to IAdminACLV0 interface. Check for
+     * null address first to avoid revert when admin has renounced ownership.
+     */
+    function adminACLAllowed(
+        address _sender,
+        address _contract,
+        bytes4 _selector
+    ) public returns (bool) {
+        return
+            owner() != address(0) &&
+            adminACLContract.allowed(_sender, _contract, _selector);
+    }
+
+    /**
+     * @notice Returns contract owner, as determined by the `superAdmin` of the underlying
+     *         `IAdminACLV0` ACL management contract.
+     * @return address Address of contract owner, as determined by underlying ACL contract
+     * @dev ref: https://docs.openzeppelin.com/contracts/4.x/api/access#Ownable
+     * @dev owner role was called `admin` prior to V3 core contract
+     */
+    function owner()
+        public
+        view
+        override(Ownable, IGenArt721CoreContractV3_Engine)
+        returns (address)
+    {
+        return adminACLContract.superAdmin();
```

Comment:
If we've renounced ownership, this will now revert. Is that the expected behavior?

Reply:
Resolving, as we've reverted this change based on convo we've had 1:1 offline about the weird implications of returning the `superAdmin`.

Also noted here, for posterity: https://github.com/ArtBlocks/artblocks-contracts/pull/361#issuecomment-1360549182